core/drand_beacon_control.go

```diff
@@ -698,7 +707,7 @@ func (bp *BeaconProcess) setupAutomaticResharing(_ context.Context, oldGroup *ke
 		Metadata:          &metadata,
 	}
 
-	metrics.ReshareStateChange(metrics.ReshareWaiting, beaconID, false)
+	metrics.ReshareStateChange(metrics.ReshareWaiting, beaconID, in.GetInfo().GetLeader())
```

Comment:
I think the leader never calls the `setupAutomaticResharing` function. But yeah, it cannot hurt to have it be dynamically tested.